helpers.py

```diff
@@ -76,12 +84,17 @@ def _pages_this_day():
 
 def _random_pages(limit):
     page_len = models.Page.objects.count()
-    indices = random.sample(xrange(page_len), limit)
-    pages = []
-    for index in indices:
-        pages.append(models.Page.objects.all()[index])
-
-    return map(_get_page_by_object, pages)
+    if page_len:
+        indices = ([i for i in xrange(page_len)]
+                    if page_len < limit
+                    else random.sample(xrange(page_len), limit))
```

Comment:
I figured out what these three lines are doing, but the combination of the ternary and loop inside the list literal made it a little tricky to follow. Wonder whether we should re-write it somehow so the loop and ternary assignment aren't combined this way :thinking: